app/notify_admin.go

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/mattermost/mattermost-server/v6/app/request"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/i18n"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+	"github.com/mattermost/mattermost-server/v6/store"
+)
+
+const lastTrialNotificationTimeStamp = "LAST_TRIAL_NOTIFICATION_TIMESTAMP"
+const lastUpgradeNotificationTimeStamp = "LAST_UPGRADE_NOTIFICATION_TIMESTAMP"
+const defaultNotifyAdminCoolOffDays = 0.0001157407407 // this is a temp change
+
+func (a *App) SaveAdminNotification(userId string, notifyData *model.NotifyAdminToUpgradeRequest) *model.AppError {
+	requiredFeature := notifyData.RequiredFeature
+	requiredPlan := notifyData.RequiredPlan
+	trial := notifyData.TrialNotification
+
+	if a.UserAlreadyNotifiedOnRequiredFeature(userId, requiredFeature) {
+		return model.NewAppError("app.SaveAdminNotification", "api.cloud.notify_admin_to_upgrade_error.already_notified", nil, "", http.StatusForbidden)
+	}
+
+	_, appErr := a.SaveAdminNotifyData(&model.NotifyAdminData{
+		UserId:          userId,
+		RequiredPlan:    requiredPlan,
+		RequiredFeature: requiredFeature,
+		Trial:           trial,
+	})
+	if appErr != nil {
+		return appErr
+	}
+
+	return nil
+}
+
+func (a *App) DoCheckForAdminNotifications(trial bool) *model.AppError {
+	ctx := request.EmptyContext(a.Srv().Log())
+	license := a.Srv().License()
+	if license == nil {
+		return model.NewAppError("DoCheckForAdminNotifications", "app.notify_admin.send_notification_post.app_error", nil, "No license found", http.StatusInternalServerError)
+	}
+
+	currentSKU := license.SkuShortName
+	workspaceName := ""
```

Comment:
@AGMETEOR how is the text showing up here with the empty workspace name?

Reply:
It reads like this "10 members of **the workspace** have requested a workspace upgrade for:"

Reply:
I should probably get rid of it since we shall be having this feature on self-hosted as well

Reply:
Hi @marianunez where did you post the suggestion to get rid of the workspace name so that I follow up?